### ISSUE TITLE
[Accessibility]: Accordion does not announce state change on click of expand/collapse all buttons on VoiceOver

### ISSUE THREAD

**Issue by RyanMunsch** [department-of-veterans-affairs/vets-design-system-documentation#6023]
### Point of contact

Darby Madewell / Accredited Representative Crew

### How severe do you think this issue is?

High

### Severity rationale

- Content being added or removed from the page must announce a change to assistive devices.
- Improper aria attributes or missing IDs must be corrected.

### URL where issue occurs

https://design.va.gov/storybook/?path=/story/uswds-va-accordion--default

### Component or pattern (if applicable)

va-accordion

### Environment details

OS: macOS 15.6.1
Browser: Chrome 142.0.7444.176
Screen Reader: VoiceOver 10
Device: MacBook Pro M2 Max

### Steps to reproduce

1.  Turn on VoiceOver
2. Navigate to [the default story for va-accordion](https://design.va.gov/storybook/?path=/story/uswds-va-accordion--default)
3. Visit "Expand all" button or "Collapse all" button and click/activate with keyboard

### Observed behavior

New expanded or collapsed state is not announce by VoiceOver

### Expected behavior / Acceptance criteria

The new state is announced by VoiceOver

### Supporting evidence (screenshots, recordings, code snippets)

https://github.com/user-attachments/assets/a9cadc69-7b07-46d4-9fd4-1ddc3291573d

### Related issues or systemic notes

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5238
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5216

### Type of issue

- [ ] Axe-core
- [ ] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [ ] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [x] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else
